File: lib/biz/spec-iot/src/test/java/site/addzero/biz/spec/iot/test/spi/CustomTelemetryNamingStrategy.java
package site.addzero.biz.spec.iot.test.spi;

import site.addzero.biz.spec.iot.IotThingRef;
import site.addzero.biz.spec.iot.spi.TelemetryTableNamingStrategy;

public final class CustomTelemetryNamingStrategy implements TelemetryTableNamingStrategy {

    @Override
    public String getName() {
        return "custom";
    }

    @Override
    public boolean supports(IotThingRef schemaRef) {
        return "product".equals(schemaRef.getKind()) && "demo-product".equals(schemaRef.getId());
    }

    @Override
    public String stableTableName(IotThingRef schemaRef) {
        return "custom_stable_demo_product";
    }

    @Override
    public String subTableName(IotThingRef schemaRef, IotThingRef sourceRef) {
        return "custom_sub_" + sourceRef.getId().replace('-', '_');
    }
}

File: lib/biz/spec-iot/src/test/kotlin/site/addzero/biz/spec/iot/test/spi/CustomTelemetryNamingStrategy.kt
package site.addzero.biz.spec.iot.test.spi

import site.addzero.biz.spec.iot.IotThingRef
import site.addzero.biz.spec.iot.spi.TelemetryTableNamingStrategy

class CustomTelemetryNamingStrategy : TelemetryTableNamingStrategy {

    override val name: String = "custom"

    override fun supports(schemaRef: IotThingRef): Boolean {
        return schemaRef.kind == "product" && schemaRef.id == "demo-product"
    }

    override fun stableTableName(schemaRef: IotThingRef): String {
        return "custom_stable_demo_product"
    }

    override fun subTableName(schemaRef: IotThingRef, sourceRef: IotThingRef): String {
        return "custom_sub_" + sourceRef.id.replace('-', '_')
    }
}
